### ISSUE TITLE
Implement `DataArray.to_dask_series()`

### ISSUE THREAD

**Issue by gcaria** [pydata/xarray#7409]
### Is your feature request related to a problem?

It'd be nice to pass from a chunk DataArray to a dask object directly

### Describe the solution you'd like

I think something along these lines should work for a 1D DataArray (although a less convoluted way might exist):

```python
def to_dask_series(self):

  index = [self.get_index(dim) for dim in self.dims][0]
  darr_index = dka.from_array(index.tolist())
  ddf = dkd.from_dask_array(self.data, columns=[self.name])
  ddf['_index']  = darr_index
  return ddf.set_index('_index').squeeze()
```

### Describe alternatives you've considered

The code should work.

### Additional context

I didn't think about the 2D case, it'd be useful to add that too to replicate what happens with `DataArray.to_pandas()`.

**Comment by dcherian** [comment]
Have you seen https://docs.xarray.dev/en/stable/generated/xarray.Dataset.to_dask_dataframe.html? I thjink we could easily add a version for DataArray that creates a temporary dataset and then calls `Dataset.to_dask_dataframe`

**Comment by gcaria** [comment]
Yes I did, but unfortunately I didn't think about the trick of converting to Dataset. My only thought then is that it'd be nice to also get a dask Series, so something like `to_dask` instead of `to_dask_dataframe`, which I guess entails adding a `DataFrame.squeeze()` somewhere in the existing `to_dask_dataframe` code.

Feel free to close this.

**Comment by dsgreen2** [comment]
Hi. I am interested in attempting to work on this issue to add the feature. Thank you.

**Comment by akanshajais** [comment]
@gcaria , Your solution looks like a reasonable approach to convert a 1D or 2D chunked DataArray to a dask DataFrame or Series, respectively.  this solution will only work if the DataArray is chunked along one or both dimensions. If the DataArray is not chunked, then calling to_dask() will return an equivalent in-memory pandas DataFrame or Series.

One potential improvement you could make is to add a check to ensure that the chunking is valid for conversion to a dask DataFrame or Series. For example, if the chunk sizes are too small, the overhead of parallelism may outweigh the benefits.

Here's an updated version of your code that includes this check:

```
import dask.dataframe as dkd
import xarray as xr
from typing import Union

def to_dask(da: xr.DataArray) -> Union[dkd.Series, dkd.DataFrame]:

    if da.data.ndim > 2:
        raise ValueError(f"Can only convert 1D and 2D DataArrays, found {da.data.ndim} dimensions")

    # Check that the chunk sizes are not too small
    min_chunk_size = 100_000  # Adjust as needed
    if any(cs < min_chunk_size for cs in da.data.chunks):
        raise ValueError("Chunk sizes are too small for conversion to dask DataFrame/Series")

    indexes = [da.get_index(dim) for dim in da.dims]
    darr_index = dkd.from_array(indexes[0], chunks=da.data.chunks[0])
    columns = [da.name] if da.data.ndim == 1 else indexes[1]
    ddf = dkd.from_dask_array(da.data, columns=columns)
    ddf[indexes[0].name] = darr_index
    return ddf.set_index(indexes[0].name).squeeze()
```
This code adds a check to ensure that the chunk sizes are not too small (in this case, we've set the minimum chunk size to 100,000). If any of the chunks have a size smaller than the minimum, then the function raises a ValueError. You can adjust the minimum chunk size as needed for your specific use case.
